### ISSUE TITLE
Which implementation of IAsyncProfilerProvider should I use in multithreaded console or service application?

### ISSUE THREAD

**Issue by LeonidVasilyev** [MiniProfiler/dotnet#198]
Readme.md references console application sample which uses `SingletonProfilerProvider`. Sample contains `TestMultiThreaded()` call. However `SingletonProfilerProvider` summary states:

> /// &lt;summary&gt;
    /// Mostly for unit testing and single-threaded apps, only allows one 
    /// instance of a &lt;see cref="MiniProfiler"/&gt; to be the &lt;see cref="MiniProfiler.Current"/&gt; one.
    /// &lt;/summary&gt;

I ran following code snippet with MiniProfiler 4.0.0-alpha6-79 and it looks like it works fine without any explicit `IAsyncProfilerProvider` setup (probably with `DefaultProfilerProvider`?):

```c#
using StackExchange.Profiling;
using System;
using System.Threading;
using System.Threading.Tasks;

namespace ConsoleApplication
{
    class Program
    {
        static void Main(string[] args)
        {
            MiniProfiler.Start();

            using (MiniProfiler.Current.Step("Cycle"))
            {
                var unused = 0;
                for(var i = 0; i < 1000; ++i)
                {
                    ++unused;
                }
            }

            using (MiniProfiler.Current.Step("Task"))
            {
                Task.Run(async () =>
                {
                    using (MiniProfiler.Current.Step("Inside task"))
                    {
                        await Task.Delay(100);
                    }
                }).Wait();
            }

            var t1 = new Thread(() =>
            {
                using (MiniProfiler.Current.Step("Inside thread"))
                {
                    Thread.Sleep(100);
                }
            });
            using (MiniProfiler.Current.Step("Thread"))
            {
                t1.Start();
                t1.Join();
            }

            var t2 = new Thread(() =>
            {
                using (MiniProfiler.Current.Step("Inside thread"))
                {
                    Thread.Sleep(100);
                }
            });
            using (MiniProfiler.Current.Step("Thread"))
            {
                t2.Start();
            }
            t2.Join();

            MiniProfiler.Stop();
            Console.WriteLine(MiniProfiler.Current.RenderPlainText());
        }
    }
}
```

Output:
>PCName at 7/25/2017 8:50:56 AM
ConsoleApplication.exe = 331.5ms
&gt; Cycle = 1.6ms
&gt; Task = 124.4ms
&gt;&gt; Inside task = 115.1ms
&gt; Thread = 101.2ms
&gt;&gt; Inside thread = 100.3ms
&gt; Thread = 0.3ms
&gt;&gt; Inside thread = 100.1ms

**Comment by NickCraver** [comment]
Yep the default is just fine - I'll leave this open as a note to add a section to the docs on this advice specifically. It's likely to be a common question for anyone converting as well.

**Comment by NickCraver** [comment]
Fixed in 19f0df715bd711a6afbe92b4fc468b6748fcf2eb by updating usage and comments on the sample :)
